File: 02-week-2--recursion/01-day-1--recursive-counting/ruby/recursive_count.rb
def recursive_count(num = 0)
  return if num >= 10

  puts num
  recursive_count(num + 1)
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# Problem: 
# Change this while loop into a recursive method
# The method should print numbers 0-9.

# count = 0

# while count < 10
#   puts count
#   count += 1
# end

File: 02-week-2--recursion/01-day-1--recursive-counting/javascript/recursive_count.js
function recursiveCount(num = 0) {
  if (num >= 10) {
    return;
  }
  recursiveCount(num + 1);
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// # Problem: 
// # Change this while loop into a recursive method
// # The method should print numbers 0-9.

// let count = 0;

// while (count < 10) {
//   console.log(count);
//   ++count;
// }